src/main/java/org/apache/commons/lang3/time/DateUtils.java

```diff
@@ -80,7 +83,7 @@
      */
     public static final int SEMI_MONTH = 1001;
 
-    private static final int[][] fields = {
+    @SuppressWarnings("assignment.type.incompatible") private static final @NonNegative int @MinLen(1) [] @MinLen(1) [] fields = { // all values are non negative
```

Comment:
You can use `@ArrayLen` when the size is fixed.

I would expect the type of this array to be `@NonNegative int @MinLen(1) [] @MinLen(8) []`.

Wrt to non-negative - are these values not annotated in the JDK? Regardless, no warning suppression should be needed here - you should mock them with a stub file or annotate them.